src/components/forms/ProfileCreation.js

```diff
@@ -0,0 +1,128 @@
+    } = this.props;
+    const userId = reduxState.profile.user;
+
+    return (
+      <CenterPaper>
+        <Formik
+          initialValues={{
+            fname: "",
+            lname: "",
+            birthdate: "",
+            phonenum: "",
+          }}
+          onSubmit={async (values) => {
+            const {
+              fname, lname, birthdate, phonenum,
+            } = values;
+
+            await initCreation(userId, fname, lname, birthdate, phonenum);
+            const response = await createProfile(userId, fname, lname, birthdate, phonenum);
+            if (response) {
+              await setId(response.id);
+              changePage();
+            }
+          }}
+        >
+          {({ values, handleChange, handleSubmit }) => (
+            <form onSubmit={handleSubmit}>
+              <h2>Create your profile</h2>
+              <TextField
+                id="fname"
+                label="First name"
+                className={classes.TextField}
+                style={{ width: "100%" }}
+                type="text"
+                name="fname"
+                autoComplete="fname"
+                onChange={handleChange}
+                value={values.fname}
+                margin="normal"
+              />
+              <TextField
+                id="lname"
+                label="Last name"
+                className={classes.TextField}
+                style={{ width: "100%" }}
+                type="text"
+                name="lname"
+                autoComplete="lname"
+                onChange={handleChange}
+                value={values.lname}
+                margin="normal"
+              />
+              <TextField
+                id="birthdate"
+                label="Birthday"
+                className={classes.TextField}
+                style={{ width: "100%" }}
+                type="date"
+                name="birthdate"
+                autoComplete="birthdate"
+                onChange={handleChange}
+                value={values.birthdate}
+                InputLabelProps={{ shrink: true }}
+                margin="normal"
+              />
+              <TextField
+                id="phonenum"
+                label="Phone number"
+                className={classes.TextField}
+                style={{ width: "100%" }}
+                type="text"
+                name="phonenum"
+                autoComplete="phonenum"
+                onChange={handleChange}
+                value={values.phonenum}
+                margin="normal"
+              />
+              <Grid item>
+                <Button color="primary" type="submit" variant="contained">
+                  Submit
+                </Button>
+              </Grid>
+            </form>
+          )}
+        </Formik>
+      </CenterPaper>
+    );
+  }
+}
+
+const componentWithStyles = withStyles(styles)(ProfileCreation);
+
+function mapStateToProps(state) {
+  return {
+    reduxState: state,
```

Comment:
this is not a good naming convention, usually you need to specify very specific, reduxState is no good

```{
    profile: state.userProfile,
    payment: state.payment
}```